internal/jsonrpc/jsonrpc.go

```diff
@@ -198,7 +198,7 @@ func CreateErrorJSONRPCResponseBodyWithRequest(message string, jsonRPCStatusCode
 	switch r := request.(type) {
 	case *SingleRequestBody:
 		response := CreateErrorJSONRPCResponseBody(message, jsonRPCStatusCode)
-		response.ID = int(r.ID)
+		response.ID = *r.ID
```

Comment:
I think we'll need to check `r.ID != nil` before deferencing, else we might get a panic.

Reply:
I've seen the graph node send out requests before w/o an ID. Specifically the web3_version method. Maybe there's others.